var/spack/repos/builtin/packages/gmap-gsnap/package.py

```diff
@@ -37,3 +37,19 @@ class GmapGsnap(AutotoolsPackage):
     version('2018-02-12', '13152aedeef9ac66be915fc6bf6464f2')
     version('2017-06-16', 'fcc91b8bdd4bf12ae3124de0c00db0c0')
     version('2014-12-28', '1ab07819c9e5b5b8970716165ccaa7da')
+
+    variant(
+        'binaries',
+        description='CPU support.',
+        values=('avx2', 'sse42', 'avx512', 'sse2'),
+        multi=True
+    )
+
+    def install(self, spec, prefix):
+        for x in ('avx2', 'sse42', 'avx512', 'sse2'):
```

Comment:
You'll probably want to use:
```python
for binary in spec.variants['binaries'].value:
```
here. Then you don't need the if-statement.

Reply:
@davydden @citibeth I seem to recall another package like this one where we wanted to run `configure`, `make`, `make install` multiple times and we managed to build each version in a separate sub-directory. Do you remember which package it was? That would allow us to still use `AutotoolsPackage` and keep each phase separate.

Reply:
i think `fftw` is one of those

Reply:
I also had a similar problem in #7571, which we currently have only in our own spack

Reply:
I think it's different from fftw because it only runs one simd each time.

Reply:
What do you mean only one simd each time?

Reply:
It can only run one simd during each build process, so need to clean then build again if want multiple cpu support.

Reply:
having built this by hand before, I think it would probably work fine to build in separate directories, and then install everything into prefix - seems that's what fftw is doing if I'm reading it right